### PR TITLE
python-pycparser: Update to 2.21, update patch

### PR DESCRIPTION
**Pull request by jefferyto**
Maintainer: me
Compile tested: armvirt-32, 2021-11-14 snapshot sdk
Run tested: none

Description:
Signed-off-by: Jeffery To <jeffery.to@gmail.com>

### CODE REVIEW

**Comment by jefferyto** [comment]
If there are no objections, I will merge this in 3 days.

File: lang/python/python-pycparser/patches/001-use-external-ply.patch [hunk 1]
```diff
@@ -13,19 +13,19 @@
  class CLexer(object):
 --- a/pycparser/c_parser.py
 +++ b/pycparser/c_parser.py
-@@ -8,7 +8,7 @@
+@@ -6,7 +6,7 @@
+ # Eli Bendersky [https://eli.thegreenplace.net/]
+ # License: BSD
```
Comment (commodo) [inline lang/python/python-pycparser/patches/001-use-external-ply.patch:18]: i'd try to send this upstream eventually;
it's been in our tree for 4-5 years now;

File: lang/python/python-pycparser/Makefile [hunk 2]
```diff
@@ -8,11 +8,11 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=python-pycparser
-PKG_VERSION:=2.20
-PKG_RELEASE:=5
+PKG_VERSION:=2.21
+PKG_RELEASE:=1
```
Comment (commodo) [inline lang/python/python-pycparser/Makefile:12]: i'd start using $(AUTORELEASE) at some point;
i'm also paranoid about it; 
but i also like the idea of the build-system taking care of this for us;

**Comment by commodo** [comment]
a few minor comments;
nothing for this PR now

**Pull request merged**